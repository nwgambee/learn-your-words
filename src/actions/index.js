export const getVocabWord = query => ({
    type: 'GET_WORD',
    query
})