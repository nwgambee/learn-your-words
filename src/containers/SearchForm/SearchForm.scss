.search-word-form {
  // font-family: 'Spartan', sans-serif;
  // border: 3px solid green;
}
