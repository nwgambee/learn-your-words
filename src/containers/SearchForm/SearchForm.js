import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getVocabWord } from '../../actions/index'
// import PropTypes from 'prop-types';
import { searchVocabWord } from '../../apiCalls/apiCalls'
import './SearchForm.scss'

class SearchForm extends Component {
    constructor() {
        super();
        this.state = { query: ''};
     }
     handleChange = e => {
        this.setState({ [e.target.name]: e.target.value})
     }
     handleError = () => {
         console.log('error');
     }
     handleSubmit = async (e) => {
         e.preventDefault();
         try {
             const result = await searchVocabWord(this.state.query)
             this.props.getVocabWord(result)
        }
        catch (e) {
            this.handleError()
        }
        this.setState({query:''});
     }
     render() {
        const { query } = this.state
         return(
             <form className='search-word-form' onSubmit={this.handleSubmit} autoComplete='off'>
                 <input 
                    className='word-input'
                    type='text'
                    placeholder="Vocab Word"
                    name="query"
                    value={query}
                    onChange={this.handleChange}
                    />
                    <button className='submit-btn' >Find Vocab Word</button>
             </form>
         )
     }
}

export const mapDispatchToProps = dispatch => ({
    getVocabWord: (query) => dispatch(getVocabWord(query))
  })

export default connect(null, mapDispatchToProps)(SearchForm);
