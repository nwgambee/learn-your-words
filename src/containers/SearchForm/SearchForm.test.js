import React from 'react';
import { getVocabWord } from '../../actions/index' 
import { SearchForm, mapStateToProps, mapDispatchToProps } from './SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {
  it('should match the snapshot', () => {
    let wrapper = shallow(<SearchForm />);
    expect(wrapper).toMatchSnapshot();
  })
  describe('mapDispatchToProps', () => {
    it('should call dispatch with a getVocabWord action when getVocabWord is called', () => {
      const mockDispatch = jest.fn();
      const action = getVocabWord('query');
      const mappedProps = mapDispatchToProps(mockDispatch)
  
      mappedProps.getVocabWord('query');
      expect(mockDispatch).toHaveBeenCalledWith(action);
    })
  })
})
