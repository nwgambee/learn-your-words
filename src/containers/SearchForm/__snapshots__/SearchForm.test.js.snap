// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<form
  autoComplete="off"
  className="search-word-form"
  onSubmit={[Function]}
>
  <input
    className="word-input"
    name="query"
    onChange={[Function]}
    placeholder="Vocab Word"
    type="text"
    value=""
  />
  <button
    className="submit-btn"
  >
    Find Vocab Word
  </button>
  <p
    className="error-message"
  />
</form>
`;

exports[`SearchForm should match the snapshot 1`] = `
<form
  autoComplete="off"
  className="search-word-form"
  onSubmit={[Function]}
>
  <input
    className="word-input"
    name="query"
    onChange={[Function]}
    placeholder="Vocab Word"
    type="text"
    value=""
  />
  <button
    className="submit-btn"
  >
    Find Vocab Word
  </button>
  <p
    className="error-message"
  />
</form>
`;
